mlir/include/mlir/Transforms/Passes.td

```diff
@@ -214,6 +214,13 @@ def SCCP : Pass<"sccp"> {
   let constructor = "mlir::createSCCPPass()";
 }
 
+def SROA : Pass<"sroa"> {
```

Comment:
This still needs some polishing :)

Reply:
Tell me if what I wrote is satisfying.

Reply:
Sounds good! Should we also add a pass flag that limits the size of the aggregates we consider for destruction (probably in terms of number of elements?).